### PR TITLE
Feature/batch-inference

### PR DESCRIPTION
**Pull request by xeon27**
# PR Type ([Feature | Fix | Documentation | Test])
Feature

## Short Description
Batch-Inference (not async) - Support to make generation and activation retrieval requests in batches

## Tests Added
...

### CODE REVIEW

**Review by markcoatsworth — APPROVED** [review]
Everything here looks good to me. The only issue as discussed on Slack is that these tests will fail in Github Actions because they're trying to connect to a service that's inaccessible behind a VPN. In the past I asked Janakan to remove tests that don't work; now I'm thinking maybe we should keep them so we can manually test in the Vaughan cluster. Let's just leave them in for now, we can decide later.

**Comment by xeon27** [comment]
Fixed the import issue, but as @markcoatsworth mentioned above the test won't work as Github doesn't have access to the service.

File: test/unit_tests/test_batch_prompting.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    _model = _client.load_model(model_name)
+    _requested_activation = ['decoder.layers.30.fc1']
+
+    def test_generation_single_prompt_str(self):
+        prompt = "What is this"
+        response = self._model.generate(prompt)
+        assert isinstance(response.generation["text"], list), f'Expected type list, found {type(response.generation["text"])}'
+        assert len(response.generation["text"]) == 1, f'Expected 1 generation, found {len(response.generation["text"])}'
+
+    def test_generation_single_prompt_list(self):
```
Comment (jwilles) [inline test/unit_tests/test_batch_prompting.py:20]: The issue with the tests is that they are integration tests and not unit tests. This is not testing the relatively simple functionality of the generate function but testing the the correctness of the entire system. We likely should be mocking the calls to the backend. See here for more discussion https://stackoverflow.com/questions/71111067/how-does-mock-testing-rest-apis-test-the-api-when-the-actual-api-is-not-called

**Comment by jwilles** [comment]
See comment above. Can we ignore integration tests until we have a solution?

**Comment by xeon27** [comment]
I think we should ignore integration tests, since these test cases are for the purpose of verifying that the responses are in the expected format. Mocking won't help in this case. @markcoatsworth - What are your thoughts on this?

**Comment by markcoatsworth** [comment]
@xeon27 an @jwilles I don't have any strong feelings here. We could just rip the integration tests out for now, since they'll still be in our git history and can be re-added anytime. If you think it'll be cleaner to ignore them for now then sure, go ahead and take them out.
Also, it looks like we have a merge conflict on `lingua/lingua_sdk.py`, can you take a look at this please? It might be related to a change I just pushed.

**Comment by markcoatsworth** [comment]
Let's deal with the tests later, we need to cut a release. Merging now.

**Pull request merged**